### PR TITLE
Fix spec

### PR DESCRIPTION
**Pull request by chuenlok**
This replaced some exact html matching code with better html element checking.

### CODE REVIEW

File: lib/tasks/wikirate.rake [hunk 1]
```diff
@@ -89,13 +100,13 @@ namespace :wikirate do
...
+        ActiveRecord::Base.connection.execute "delete from cards where type_id = '#{Card::WikirateTopicID}' and id not in ( #{topic_ids[0...-1]} )"
 
         # delete all webpage++link
-        ActiveRecord::Base.connection.execute "delete ca from cards ca inner join cards le ON ca.left_id = le.id where le.type_id in ('631','651','699','974','1010','1109','1591','1638','1690','2207','2317','2327','2754','2755','2813','2995','4010','4030') and ca.id not in  ( #{card_to_be_kept[0...-1]} )" 
-        ActiveRecord::Base.connection.execute "delete from cards where type_id in ('631','699','974','1109','1591','1638','1690','2207','2317','2327','2754','2755','2813','2995','4010','4030') and id not in ( #{card_to_be_kept[0...-1]} )"
+        ActiveRecord::Base.connection.execute "delete ca from cards ca inner join cards le ON ca.left_id = le.id where le.type_id in (#{type_ids[0...-1]}) and ca.id not in  ( #{card_to_be_kept[0...-1]} )" 
+        ActiveRecord::Base.connection.execute "delete from cards where type_id in (#{type_ids_without_company_and_topic[0...-1]}) and id not in ( #{card_to_be_kept[0...-1]} )"
```
Comment (ethn) [inline lib/tasks/wikirate.rake]: Ahh, MUCH more readable!

**Comment by ethn** [comment]
good stuff.  will merge in.  very close now!

File: lib/tasks/wikirate.rake (hunk 1 above)
Comment (xithan) [inline lib/tasks/wikirate.rake]: I guess the `[0..-1]` is there to get rid of the comma at the end? -1 includes the end of the string so it should say `[0..-2]` or maybe easier to read: collect the ids in an array and use `join ','`. Then you get rid of the initialization statement as well `type_ids = %w{ ... }.map { ... }.join ','`

Comment (chuenlok) [inline lib/tasks/wikirate.rake]: let me try your comment :+1:

**Pull request closed without merging**